duckduckgo/Android
# ViewBinding activities migration

Task/Issue URL: https://app.asana.com/0/414730916066338/1200552671375508/f
Tech Design URL: 
CC: 

**Description**:
This PR migrates the current activities to use ViewBinding:

- SystemSearchActivity
- OnboardingActivity
- BrowserActivity
- PrivacyDashboardActivity
- ScorecardActivity
- TrackerNetworksActivity
- PrivacyPracticesActivity
- BrokenSitesActivity
- SurveyActivity
- AddWidgetInstructionsActivity
- FireproofWebsitesActivity
- LocationPermissionsActivity
- ChangeIconActivity
- WhiteListActivity
- GlobalPrivacyControlActivity


**Steps to test this PR**:
Let's perform a smoke test through all the migrated activities:

1. Fresh install
1. Go through onboarding
2. visit any website
3. Go to privacy dashboard
4. Click on the header (grey area where the privacy grade is shown)
5. Ensure you navigate to privacy grade
6. Go back, and click on trackers blocked
7. Ensure you navigate to tracker networks list
8. Go back, Click on Unprotected sites
9. Ensure you navigate to unprotected sites (list should be empty)
10. add a site manually (like example.com)
11. go back, and disable privacy protection
12. Click again on unprotected sites
13. Ensure the site you added manually and the current sites (we've just disabled privacy protection) are listed 
14. click on privacy practices
15. Ensure you navigate to privacy practices
16. click on Report broken site
17. Ensure you navigate to report brokens site and submit any report
18. Go back to the browser screen
19. Fireproof the site 
20. Go to settings -> fireproof sites
21. Ensure you navigate to Fireproof sites and list contains the site we've just fireproofed
22. Go to the browser screen
23. search "restaurants near me"
24. when location prompt appears, select "always" for this site (you may need to enable location permissions manually for the prompt to appear)
25. Go to settings -> Location
26. Ensure you navigate to Location screen and ddg is on the list
25. Go to Settings -> App Icon
26. Ensure you navigate to App icon and change the app icon
27. Go to Settings -> Global privacy control
28. Ensure you navigate to GPC screen, disable and enable GPC toogle
29. Go to you home screen, add DDG widget
30. when added, click on the widget
31. Ensure you navigate to SystemSearchActivity, perform any search from there
32. Ensure autocomplete results appeared as expected


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
